java-spiffe-core/src/main/java/io/spiffe/workloadapi/JwtSource.java

```diff
@@ -0,0 +1,256 @@
+package io.spiffe.workloadapi;
+
+import io.spiffe.bundle.BundleSource;
+import io.spiffe.bundle.jwtbundle.JwtBundle;
+import io.spiffe.bundle.jwtbundle.JwtBundleSet;
+import io.spiffe.exception.BundleNotFoundException;
+import io.spiffe.exception.JwtSourceException;
+import io.spiffe.exception.JwtSvidException;
+import io.spiffe.exception.SocketEndpointAddressException;
+import io.spiffe.spiffeid.SpiffeId;
+import io.spiffe.spiffeid.TrustDomain;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+import lombok.val;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import io.spiffe.bundle.x509bundle.X509Bundle;
+import io.spiffe.svid.jwtsvid.JwtSvid;
+import io.spiffe.svid.jwtsvid.JwtSvidSource;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+
+import static io.spiffe.workloadapi.internal.ThreadUtils.await;
+
+/**
+ * Represents a source of SPIFFE JWT SVIDs and JWT bundles maintained via the Workload API.
+ */
+@Log
+public class JwtSource implements JwtSvidSource, BundleSource<JwtBundle>, Closeable {
```

Comment:
I decided to not provide an asynchronous method with this PR because of some complexities that entails returning a CompletableFuture, if the user decides to cancel it, the JwtSource (and underlying channels, etc) should be closed properly, what proved to be not straightforward. We could handle it separately in another issue.